pkg/plugins/golang/deploy-image/v1alpha1/scaffolds/internal/templates/api/types.go

```diff
@@ -117,10 +122,18 @@ type {{ .Resource.Kind }}Status struct {
 // {{ .Resource.Kind }} is the Schema for the {{ .Resource.Plural }} API
 type {{ .Resource.Kind }} struct {
 	metav1.TypeMeta   ` + "`" + `json:",inline"` + "`" + `
+
+	// metadata is a standard object metadata.
+	// +optional
 	metav1.ObjectMeta ` + "`" + `json:"metadata,omitempty"` + "`" + `
 
-	Spec   {{ .Resource.Kind }}Spec   ` + "`" + `json:"spec,omitempty"` + "`" + `
-	Status {{ .Resource.Kind }}Status ` + "`" + `json:"status,omitempty"` + "`" + `
+	// spec defines the desired state of {{ .Resource.Kind }}.
+	// +required
+	Spec   {{ .Resource.Kind }}Spec   ` + "`" + `json:"spec"` + "`" + `
+
+	// status defines the observed state of {{ .Resource.Kind }}.
+	// +optional
+	Status *{{ .Resource.Kind }}Status ` + "`" + `json:"status,omitempty"` + "`" + `
```

Comment:
@JoelSpeed  and @everettraven 

I understand that by default, the best practice would be to ensure that the spec is required.
If not, please let me know.